File: rust/src/models/rest_error_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::error_code::ErrorCode;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct RestErrorResponse {
    pub code: ErrorCode,

    pub message: String,
}

impl RestErrorResponse {
    pub fn new(code: ErrorCode, message: String) -> Self {
        Self { code, message }
    }
}

File: browser/src/models/restErrorResponse.ts
import { type ErrorCode, ErrorCodeSerializer } from "./errorCode";

export interface RestErrorResponse {
  code: ErrorCode;

  message: string;
}

export const RestErrorResponseSerializer = {
  _fromJsonObject(object: any): RestErrorResponse {
    return {
      code: ErrorCodeSerializer._fromJsonObject(object["code"]),
      message: object["message"],
    };
  },

  _toJsonObject(self: RestErrorResponse): any {
    return {
      code: ErrorCodeSerializer._toJsonObject(self.code),
      message: self.message,
    };
  },
};
